R/agg_scale.R

```diff
@@ -0,0 +1,879 @@
+#' @title Aggregate/Scale a detailed level of a variable to an aggregate level
+#'
+#' @description Aggregate counts or probabilities from a detailed level of a
+#' variable to an aggregate level or scale the detailed level values so that the
+#' detailed level aggregated together equals the aggregate level.
+#'
+#' @param dt \[`data.table()`\]\cr
+#'   Data to be aggregated or scaled.
+#' @param id_cols \[`character()`\]\cr
+#'   ID columns that uniquely identify each row of `dt`.
+#' @param value_cols \[`character()`\]\cr
+#'   Value columns that should be aggregated.
+#' @param col_type \[`character(1)`\]\cr
+#'   The type of variable that is being aggregated or scaled over. Can be either
+#'   'categorical' or 'interval'.
+#' @param col_stem \[`character(1)`\]\cr
+#'   The name of the variable to be aggregated or scaled over. If aggregating an
+#'   'interval' variable should not include the '_start' or '_end' suffix.
+#' @param mapping \[`data.table()`\]\cr
```

Comment:
what columns? also -- do we need helper functions to format GBD shared database location and age tables into the correct "mapping" arg format?

Reply:
The required columns are described in the `details` section. Per @spencerpease earlier feedback tried to keep some of the details out of this section.

Yes definitely will want some helper functions